.github/workflows/nightly-emscripten.yml

```diff
@@ -0,0 +1,120 @@
+name: Emscripten nightly build + solc-bin push
+
+on:
+  schedule:
+    # Run once a day, at midnight
+    - cron: '0 0 * * *'
+
+jobs:
+  build-emscripten:
+    runs-on: ubuntu-latest
+    outputs:
+      nightly-version: ${{ env.NIGHTLY_VERSION }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'ethereum/solidity'
+          ref: 'develop'
+
+      - name: Build soljson.js
+        run: |
+          # Note that this script will spawn and build inside a docker image (which works just fine in github actions).
+          scripts/build_emscripten.sh
+
+      - name: Get nightly version
+        id: nightly-version
+        run: |
+          last_commit_timestamp=$(git log -1 --date=iso --format=%ad HEAD)
+          last_commit_date=$(date --date="$last_commit_timestamp" --utc +%Y.%-m.%-d)
+          last_commit_hash=$(git rev-parse --short=8 HEAD)
+          solidity_version=$(scripts/get_version.sh)
+          nightly_version="v${solidity_version}-nightly.${last_commit_date}+commit.${last_commit_hash}"
+
+          echo "::set-env name=NIGHTLY_VERSION::${nightly_version}"
+
+      - name: Upload soljson.js as an artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: soljson.js
+          path: upload/soljson.js
+
+  commit-soljson:
+    runs-on: ubuntu-latest
+    needs: build-emscripten
+
+    env:
+      COMMITTER_NAME: emscripten nightly action
+      # TODO: Set a proper e-mail address
+      COMMITTER_EMAIL: solidity@example.com
+      TARGET_BRANCH: master
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      NIGHTLY_VERSION: ${{ needs.build-emscripten.outputs.nightly-version }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ env.TARGET_BRANCH }}
+
+      - name: Make sure that the working copy is clean
+        run: |
+          echo "::debug::$(git status)"
+          test -z "$(git status --porcelain)" || exit 1
+
+      - name: Check if the nightly already exists
+        run: |
+          version_already_exists=$(git ls-files "bin/soljson-$NIGHTLY_VERSION.js" | sed q1 || echo true && true)
```

Comment:
Why not use `[ -e "file" ]`? This is a checkout with working copy, right?

Reply:
Actually, I think we even *have* to use ``[ -e "file" ]`` here.
The ``git ls-files`` thing was meant to check if there is a uncommited new file in the repo, after copying it over, but we're now checking *before* copying over (which does make more sense), but then ``git ls-files`` like this here won't work.

Reply:
@ekpyron Both `[ -e "file" ]` and `git ls-files` would work. Note that I dropped the `--other` option so I'm actually checking for committed files.

Both will have the same effect as long as there are no uncommitted files - and the previous step enforces that. But semantically the `git ls-files` check is closer to what we really mean. If, for whatever reason, the file happens to be on disk but not in the repo, it does not mean that the nightly has already been built and we should not skip the remaining steps.

As a bonus, `git ls-files` works with and without a checkout. One less thing to change if the repo starts growing way beyond the original 10 GB limit and at some point we decide to go back to a checkout-less version.

Reply:
Hm, the previous scripts stopped if there was already a version with the same date - maybe we should also be more strict here?

Reply:
Sounds like a good idea. Though the check might break if we ever change the date format (and I wanted to propose just that ;))

Reply:
Checks added.